nuxeo-quota-core/src/main/java/org/nuxeo/ecm/quota/size/DocumentsSizeUpdater.java

```diff
@@ -332,6 +311,40 @@ protected void processDocumentRestored(CoreSession session, DocumentModel doc) {
         updateDocumentAndAncestors(session, doc, size, size + versionsSize, 0, versionsSize);
     }
 
+    @Override
+    protected void processDocumentTrashed(CoreSession session, DocumentModel doc) {
+        QuotaAware quotaDoc = doc.getAdapter(QuotaAware.class);
+        if (quotaDoc == null) {
+            return;
+        }
+        long size = quotaDoc.getInnerSize();
+        if (log.isTraceEnabled()) {
+            if (quotaDoc.getDoc().isFolder()) {
+                log.trace(quotaDoc.getDoc().getPathAsString() + " is a folder, just inner size (" + size
+                        + ") taken into account for trash size");
+            }
+        }
+        // constraints check not needed, since the documents stays in the same folder
+        updateDocumentAndAncestors(session, doc, 0, 0, size, 0);
```

Comment:
I dont see the point of splitting `processDocumentTrashOp` into two methods when the have essentially the same content. I think it would be simpler to have changed the `String transition` parameter into a `boolean isTrash` and keep the factored code.